src/category_theory/comma.lean

```diff
@@ -229,6 +230,56 @@ variables {Y : T} {f : X ⟶ Y} {U V : over X} {g : U ⟶ V}
 @[simp] lemma map_map_left : ((map f).map g).left = g.left := rfl
 end
 
+section iterated_slice
+variables (f : over X)
+
+/-- Given f : Y ⟶ X, this is the obvious functor from (T/X)/f to T/Y -/
+@[reducible]
+def iterated_slice_forward : over f ⥤ over f.left :=
+{ obj := λ α, over.mk α.hom.left,
+  map := λ α β κ, over.hom_mk κ.left.left (begin rw auto_param_eq, rw ← over.w κ, refl end)}
+
+/-- Given f : A ⟶ B, this is the obvious functor from T/Y to (T/X)/f -/
+@[reducible]
+def iterated_slice_backward : over f.left ⥤ over f :=
+{ obj := λ g, over.mk (over.hom_mk g.hom (by simp) : over.mk (g.hom ≫ f.hom) ⟶ _),
+  map := λ g h α, @over.hom_mk _ _ f
+              (over.mk (@over.hom_mk _ _ X (over.mk (g.hom ≫ f.hom)) f g.hom (by simp) : _ ⟶ f))
+              (over.mk (@over.hom_mk _ _ X (over.mk (h.hom ≫ f.hom)) f h.hom (by simp) : _ ⟶ f))
+              (over.hom_mk α.left (over.w_assoc α f.hom)) (over.over_morphism.ext (over.w α)) }
```

Comment:
All that `@`s here suggest that maybe we made too many of the arguments of `over.hom_mk` implicit. Do you want to experiment?

Reply:
Turned out I didn't need lots of them!